remail/email_api/service.py

```
@@ -0,0 +1,519 @@
+            return func(self, *args, **kwargs)
+        except ee.EmailError as e:
+            raise e
+        except ValueError as e:
+            if "is not an email address" in str(e):
+                raise ee.InvalidLoginData() from None
+            else:
+                raise ee.UnknownError(f"An unexpected error occurred: {str(e)}") from e
+        except INVALIDLOGINDATA:
+            raise ee.InvalidLoginData() from None
+        except CONNECTIONFAIL:
+            raise ee.ServerConnectionFail()
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)